### PR TITLE
[SIL-86] Add user data source tests

### PR DESCRIPTION
**Pull request by kezc**
- Added user data source tests

### CODE REVIEW

File: app/src/test/java/com/appunite/loudius/network/datasource/UserDataSourceTest.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                id = 1,
+                login = "exampleUser",
+            )
+        )
+
+        Assertions.assertEquals(expected, actualResponse) { "Data should be valid" }
+    }
+
+    @Test
+    fun `Given incorrect access token WHEN processing request THEN return failure with Network error`() =
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/network/datasource/UserDataSourceTest.kt]: **Issue:** Name states it returns Network error but tests state it expects UnknownError.

Comment (Krzysiudan) [reply app/src/test/java/com/appunite/loudius/network/datasource/UserDataSourceTest.kt]: The name is not corrected here.

**Review by Krzysiudan — APPROVED** [review]
Approved but one name is not corrected yet 😅

**Review by nowakweronika — COMMENTED** [review]
Conflict with `PullRequestsNetworkDataSourceTest.kt` :)

**Pull request merged**